app/controllers/nonsipp/sharesdisposal/ReportedSharesDisposalListController.scala

```diff
@@ -0,0 +1,312 @@
+                    BadRequest(
+                      view(
+                        errors,
+                        viewModel(
+                          srn,
+                          mode,
+                          page,
+                          indexes,
+                          numberOfDisposals,
+                          maxPossibleNumberOfDisposals,
+                          request.userAnswers
+                        )
+                      )
+                    )
+                )
+                .merge,
+            answer =>
+              Redirect(navigator.nextPage(ReportedSharesDisposalListPage(srn, answer), mode, request.userAnswers))
+          )
+      }
+    }.merge
+  }
+
+  private def getDisposals(srn: Srn)(implicit request: DataRequest[_]): Either[Result, Map[Max5000, List[Max50]]] =
+    request.userAnswers
+      .map(SharesDisposalCompletedPages(srn))
+      .filter(_._2.nonEmpty)
+      .map {
+        case (key, sectionCompleted) =>
+          val maybeSharesIndex: Either[Result, Max5000] =
+            refineStringIndex[Max5000.Refined](key).getOrRecoverJourney
+
+          val maybeDisposalIndexes: Either[Result, List[Max50]] =
+            sectionCompleted.keys.toList
+              .map(refineStringIndex[Max50.Refined])
+              .traverse(_.getOrRecoverJourney)
+
+          for {
+            sharesIndex <- maybeSharesIndex
+            disposalIndexes <- maybeDisposalIndexes
+          } yield (sharesIndex, disposalIndexes)
+      }
+      .toList
+      .sequence
+      .map(_.toMap)
+
+  private def getSharesDisposalsWithIndexes(srn: Srn, disposals: Map[Max5000, List[Max50]])(
+    implicit request: DataRequest[_]
+  ): Either[Result, List[((Max5000, List[Max50]), SectionCompleted)]] =
+    disposals
+      .map {
+        case indexes @ (shareIndex, _) =>
+          request.userAnswers
+            .get(SharesCompleted(srn, shareIndex))
+            .getOrRecoverJourney
+            .map(sharesDisposal => (indexes, sharesDisposal))
+      }
+      .toList
+      .sequence
+}
+
+object ReportedSharesDisposalListController {
+  def form(formProvider: YesNoPageFormProvider): Form[Boolean] =
+    formProvider(
+      "sharesDisposal.reportedSharesDisposalList.radios.error.required"
+    )
+
+  private def rows(
+    srn: Srn,
+    mode: Mode,
+    sharesDisposalsWithIndexes: List[((Max5000, List[Max50]), SectionCompleted)],
+    userAnswers: UserAnswers
+  ): List[ListRow] =
+    sharesDisposalsWithIndexes.flatMap {
+      case ((shareIndex, disposalIndexes), sharesDisposal) =>
+        disposalIndexes.map { disposalIndex =>
+          val sharesDisposalData = SharesDisposalData(
+            shareIndex,
+            disposalIndex,
+            userAnswers.get(TypeOfSharesHeldPage(srn, shareIndex)).get,
+            userAnswers.get(CompanyNameRelatedSharesPage(srn, shareIndex)).get,
+            userAnswers.get(HowWereSharesDisposedPage(srn, shareIndex, disposalIndex)).get
```

Comment:
This line errors out when a share disposal was previously removed. There may be additional page data added since the removal was implemented, so the removal is now incomplete, i.e. it doesn't remove all the data